printing_auto_base/models/printing_auto.py

```diff
@@ -0,0 +1,139 @@
+    def _check_data_source(self):
+        for rec in self:
+            if rec.data_source == "report" and not rec.report_id:
+                raise UserError(_("Report is not set"))
+            if rec.data_source == "attachment" and (
+                not rec.attachment_domain or rec.attachment_domain == "[]"
+            ):
+                raise UserError(_("Attachment domain is not set"))
+
+    def _get_behaviour(self):
+        if self.printer_id:
+            result = {"printer": self.printer_id}
+            if self.printer_tray_id:
+                result["tray"] = self.printer_tray_id.system_name
+            return result
+        if self.data_source == "report":
+            return self.report_id.behaviour()
+        return self.env["ir.actions.report"]._get_user_default_print_behaviour()
+
+    def _get_record(self, record):
+        if self.record_change:
+            try:
+                return safe_eval(f"obj.{self.record_change}", {"obj": record})
+            except Exception as e:
+                raise ValidationError(
+                    _("The Record change could not be applied because: %s") % str(e)
+                ) from e
+        return record
+
+    def _check_condition(self, record):
+        domain = safe_eval(self.condition, {"env": self.env})
+        return record.filtered_domain(domain)
+
+    def _get_content(self, record):
+        generate_data_func = getattr(
+            self, f"_generate_data_from_{self.data_source}", None
+        )
+        if generate_data_func:
+            record = self._get_record(record)
+            return generate_data_func(record)
+        return []
+
+    def _prepare_attachment_domain(self, record):
+        domain = safe_eval(self.attachment_domain)
+        record_domain = [
+            ("res_id", "=", record.id),
+            ("res_model", "=", record._name),
+        ]
+        return expression.AND([domain, record_domain])
+
+    def _generate_data_from_attachment(self, record):
+        domain = self._prepare_attachment_domain(record)
+        attachments = self.env["ir.attachment"].search(domain)
+        if not attachments:
+            raise ValidationError(_("No attachment was found."))
+        return [base64.b64decode(a.datas) for a in attachments]
+
+    def _generate_data_from_report(self, record):
+        self.ensure_one()
+        data, _ = self.report_id.with_context(must_skip_send_to_printer=True)._render(
+            record.id
+        )
+        return [data]
+
+    def do_print(self, record):
+        self.ensure_one()
+        record.ensure_one()
+
+        behaviour = self._get_behaviour()
+        printer = behaviour["printer"]
+        if not printer:
+            raise UserError(
+                _("No printer configured to print this {}.").format(self.name)
+            )
+
+        if self.nbr_of_copies <= 0:
+            return (printer, 0)
+        if not self._check_condition(record):
+            return (printer, 0)
```

Comment:
```suggestion
        behaviour = self._get_behaviour()
        printer = behaviour["printer"]
        
        if self.nbr_of_copies <= 0:
            return (printer, 0)
        if not self._check_condition(record):
            return (printer, 0)
        if not printer:
            raise UserError(
                _("No printer configured to print this {}.").format(self.name)
            )
```
The check if something needs to be printed should be before the printer check

Reply:
done